Non deterministic tests failure

It seems to be related to commit: https://github.com/odoo/owl/commit/7e4baf668afc3a7d6e2f2265759c0090e2b5408d
```zsh
❯ npm run test 

> @odoo/owl@1.0.11 test /home/boi/WORK/owl
> jest

 FAIL  tests/context.test.ts
  ● Context › useContext hook is reactive, for one component

    expect(received).toBe(expected) // Object.is equality

    Expected: "<div>321</div>"
    Received: "<div>123</div>"

      54 |     test.contextObj.value = 321;
      55 |     await nextTick();
    > 56 |     expect(fixture.innerHTML).toBe("<div>321</div>");
         |                               ^
      57 |   });
      58 | 
      59 |   test("two components can subscribe to same context", async () => {

      at Object.<anonymous> (tests/context.test.ts:56:31)

 PASS  tests/misc/portal.test.ts
 PASS  tests/component/slots.test.ts
 PASS  tests/qweb/qweb.test.ts
 PASS  tests/component/un_mounting.test.ts
 FAIL  tests/store_hooks.test.ts
  ● connecting a component to store › dispatch an action

    expect(received).toBe(expected) // Object.is equality

    Expected: "<div>1</div>"
    Received: "<div>0</div>"

      1162 |     expect(res).toBe(1);
      1163 |     await nextTick();
    > 1164 |     expect(fixture.innerHTML).toBe("<div>1</div>");
           |                               ^
      1165 |   });
      1166 | });
      1167 | 

      at Object.<anonymous> (tests/store_hooks.test.ts:1164:31)

 PASS  tests/component/async.test.ts
 PASS  tests/hooks.test.ts
 PASS  tests/animations.test.ts
 PASS  tests/component/class_style.test.ts
 PASS  tests/router/router.test.ts
 PASS  tests/component/error_handling.test.ts
 PASS  tests/qweb/qweb_expressions.test.ts
 PASS  tests/misc/async_root.test.ts
 PASS  tests/component/props_validation.test.ts
 PASS  tests/tooling/debug_script_1.test.ts
 PASS  tests/router/route_component.test.ts
 PASS  tests/tooling/debug_script_3.test.ts
 PASS  tests/component/styles.test.ts
 PASS  tests/router/link.test.ts
 PASS  tests/store.test.ts
 PASS  tests/tooling/debug_script_2.test.ts
 PASS  tests/utils.test.ts
 PASS  tests/vdom.test.ts
 PASS  tests/core/observer.test.ts
 PASS  tests/tooling/debug_script_4.test.ts
 PASS  tests/tooling/doc_link_checker.test.ts
 PASS  tests/core/event_bus.test.ts
 PASS  tests/component/component.test.ts (8.034s)

Summary of all failing tests
 FAIL  tests/context.test.ts
  ● Context › useContext hook is reactive, for one component

    expect(received).toBe(expected) // Object.is equality

    Expected: "<div>321</div>"
    Received: "<div>123</div>"

      54 |     test.contextObj.value = 321;
      55 |     await nextTick();
    > 56 |     expect(fixture.innerHTML).toBe("<div>321</div>");
         |                               ^
      57 |   });
      58 | 
      59 |   test("two components can subscribe to same context", async () => {

      at Object.<anonymous> (tests/context.test.ts:56:31)

 FAIL  tests/store_hooks.test.ts
  ● connecting a component to store › dispatch an action

    expect(received).toBe(expected) // Object.is equality

    Expected: "<div>1</div>"
    Received: "<div>0</div>"

      1162 |     expect(res).toBe(1);
      1163 |     await nextTick();
    > 1164 |     expect(fixture.innerHTML).toBe("<div>1</div>");
           |                               ^
      1165 |   });
      1166 | });
      1167 | 

      at Object.<anonymous> (tests/store_hooks.test.ts:1164:31)


Test Suites: 2 failed, 27 passed, 29 total
Tests:       2 failed, 3 skipped, 765 passed, 770 total
Snapshots:   263 passed, 263 total
Time:        10.68s
Ran all test suites.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @odoo/owl@1.0.11 test: `jest`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @odoo/owl@1.0.11 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/boi/.npm/_logs/2020-10-16T09_03_10_532Z-debug.log
```

```log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'test' ]
2 info using npm@6.14.7
3 info using node@v14.13.1
4 verbose run-script [ 'pretest', 'test', 'posttest' ]
5 info lifecycle @odoo/owl@1.0.11~pretest: @odoo/owl@1.0.11
6 info lifecycle @odoo/owl@1.0.11~test: @odoo/owl@1.0.11
7 verbose lifecycle @odoo/owl@1.0.11~test: unsafe-perm in lifecycle true
8 verbose lifecycle @odoo/owl@1.0.11~test: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/boi/WORK/owl/node_modules/.bin:/home/boi/.yarn/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/boi/.local/bin/statusbar:/home/boi/.local/bin/__pycache__:/home/boi/.local/bin/cron:/home/boi/.local/bin/i3cmds:/home/boi/.local/bin/
9 verbose lifecycle @odoo/owl@1.0.11~test: CWD: /home/boi/WORK/owl
10 silly lifecycle @odoo/owl@1.0.11~test: Args: [ '-c', 'jest' ]
11 silly lifecycle @odoo/owl@1.0.11~test: Returned: code: 1  signal: null
12 info lifecycle @odoo/owl@1.0.11~test: Failed to exec test script
13 verbose stack Error: @odoo/owl@1.0.11 test: `jest`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:314:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:314:20)
13 verbose stack     at maybeClose (internal/child_process.js:1047:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
14 verbose pkgid @odoo/owl@1.0.11
15 verbose cwd /home/boi/WORK/owl
16 verbose Linux 5.8.13-zen1-2-zen
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "test"
18 verbose node v14.13.1
19 verbose npm  v6.14.7
20 error code ELIFECYCLE
21 error errno 1
22 error @odoo/owl@1.0.11 test: `jest`
22 error Exit status 1
23 error Failed at the @odoo/owl@1.0.11 test script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

--- a/tests/animations.test.ts
+++ b/tests/animations.test.ts
@@ -1,15 +1,15 @@
 import { Component, Env } from "../src/component/component";
+import { useRef, useState } from "../src/hooks";
 import { QWeb } from "../src/qweb/index";
-import { useState, useRef } from "../src/hooks";
 import { xml } from "../src/tags";
 import {
   makeDeferred,
-  makeTestFixture,
   makeTestEnv,
+  makeTestFixture,
+  nextFrame,
   patchNextFrame,
   renderToDOM,
   unpatchNextFrame,
-  nextTick,
 } from "./helpers";
 
 //------------------------------------------------------------------------------
@@ -420,29 +420,29 @@ describe("animations", () => {
 
     widget.state.flag = true;
 
-    await nextTick();
+    await nextFrame();
     widget.el!.querySelector("span")!.dispatchEvent(new Event("transitionend"));
     expect(fixture.innerHTML).toBe('<div><span class="">blue</span></div>');
     expect(QWeb.utils.transitionInsert).toBeCalledTimes(1);
 
     widget.state.flag = false;
-    await nextTick();
+    await nextFrame();
     expect(fixture.innerHTML).toBe(
       '<div><span class="chimay-leave-active chimay-leave-to" data-owl-key="__3__">blue</span></div>'
     );
     expect(QWeb.utils.transitionInsert).toBeCalledTimes(1);
 
     widget.state.flag = true;
-    await nextTick();
+    await nextFrame();
     expect(fixture.innerHTML).toBe(
       '<div><span class="chimay-enter-active chimay-enter-to" data-owl-key="__3__">blue</span></div>'
     );
     expect(QWeb.utils.transitionInsert).toBeCalledTimes(2);
 
     widget.state.flag = false;
-    await nextTick();
+    await nextFrame();
     widget.state.flag = true;
-    await nextTick();
+    await nextFrame();
 
     expect(QWeb.utils.transitionInsert).toBeCalledTimes(3);
     widget.el!.querySelector("span")!.dispatchEvent(new Event("transitionend"));

--- a/tests/helpers.ts
+++ b/tests/helpers.ts
@@ -37,8 +37,13 @@ export function nextMicroTick(): Promise<void> {
 }
 
 export async function nextTick(): Promise<void> {
-  await new Promise((resolve) => scheduler.requestAnimationFrame(resolve));
   await new Promise((resolve) => setTimeout(resolve));
+  await new Promise((resolve) => scheduler.requestAnimationFrame(resolve));
+}
+
+export async function nextFrame(): Promise<void> {
+  await new Promise((resolve) => scheduler.requestAnimationFrame(resolve));
+  await new Promise((resolve) => scheduler.requestAnimationFrame(resolve));
 }
 
 export function makeTestFixture() {
@@ -127,7 +132,7 @@ export function renderToString(
 // is useful for animations tests, as we hook before repaints to trigger
 // animations (thanks to requestAnimationFrame). Patching nextFrame allows to
 // simulate calls to this hook. One must not forget to unpatch afterwards.
-let nextFrame = QWeb.utils.nextFrame;
+let _nextFrame = QWeb.utils.nextFrame;
 export function patchNextFrame(f: Function) {
   QWeb.utils.nextFrame = (cb: () => void) => {
     setTimeout(() => f(cb));
@@ -135,7 +140,7 @@ export function patchNextFrame(f: Function) {
 }
 
 export function unpatchNextFrame() {
-  QWeb.utils.nextFrame = nextFrame;
+  QWeb.utils.nextFrame = _nextFrame;
 }
 
 export async function editInput(input: HTMLInputElement | HTMLTextAreaElement, value: string) {

